File: backend/src/main/java/com/foodya/foodya_backend/restaurant/dto/MenuItemResponse.java
package com.foodya.foodya_backend.restaurant.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Menu item details response")
public class MenuItemResponse {

    @Schema(description = "Menu item ID", example = "1")
    private UUID id;

    @Schema(description = "Restaurant ID", example = "1")
    private UUID restaurantId;

    @Schema(description = "Restaurant name", example = "The Italian Corner")
    private String restaurantName;

    @Schema(description = "Menu item name", example = "Margherita Pizza")
    private String name;

    @Schema(description = "Description", example = "Classic Italian pizza")
    private String description;

    @Schema(description = "Price", example = "12.99")
    private Double price;

    @Schema(description = "Image URL")
    private String imageUrl;

    @Schema(description = "Category", example = "Main Course")
    private String category;

    @Schema(description = "Is available?", example = "true")
    private Boolean isAvailable;

    @Schema(description = "Is active? (soft delete)", example = "true")
    private Boolean isActive;

    @Schema(description = "Preparation time in minutes", example = "15")
    private Integer preparationTime;

    @Schema(description = "Calories", example = "800")
    private Integer calories;

    @Schema(description = "Is vegetarian?", example = "true")
    private Boolean isVegetarian;

    @Schema(description = "Is vegan?", example = "false")
    private Boolean isVegan;

    @Schema(description = "Is gluten-free?", example = "false")
    private Boolean isGlutenFree;

    @Schema(description = "Is spicy?", example = "false")
    private Boolean isSpicy;

    @Schema(description = "Total orders count", example = "45")
    private Integer orderCount;

    @Schema(description = "Created timestamp")
    private LocalDateTime createdAt;

    @Schema(description = "Last updated timestamp")
    private LocalDateTime updatedAt;
}

File: mobile/app/src/main/java/com/example/foodya/data/model/MenuItemResponse.kt
package com.example.foodya.data.model

import com.example.foodya.domain.model.Food
import com.google.gson.annotations.SerializedName

data class MenuItemResponse(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("restaurantId")
    val restaurantId: String,
    
    @SerializedName("restaurantName")
    val restaurantName: String,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("description")
    val description: String? = null,
    
    @SerializedName("price")
    val price: Double,
    
    @SerializedName("imageUrl")
    val imageUrl: String? = null,
    
    @SerializedName("category")
    val category: String,
    
    @SerializedName("isAvailable")
    val isAvailable: Boolean = true,
    
    @SerializedName("isActive")
    val isActive: Boolean = true,
    
    @SerializedName("preparationTime")
    val preparationTime: Int? = null,
    
    @SerializedName("calories")
    val calories: Int? = null,
    
    @SerializedName("isVegetarian")
    val isVegetarian: Boolean = false,
    
    @SerializedName("isVegan")
    val isVegan: Boolean = false,
    
    @SerializedName("isGlutenFree")
    val isGlutenFree: Boolean = false,
    
    @SerializedName("isSpicy")
    val isSpicy: Boolean = false,
    
    @SerializedName("orderCount")
    val orderCount: Int? = 0,
    
    @SerializedName("createdAt")
    val createdAt: String? = null,
    
    @SerializedName("updatedAt")
    val updatedAt: String? = null
) {
    fun toDomain(): Food {
        return Food(
            id = id,
            restaurantId = restaurantId,
            restaurantName = restaurantName,
            name = name,
            description = description ?: "",
            price = price,
            imageUrl = imageUrl ?: "",
            category = category,
            isAvailable = isAvailable,
            isActive = isActive,
            preparationTime = preparationTime,
            calories = calories,
            isVegetarian = isVegetarian,
            isVegan = isVegan,
            isGlutenFree = isGlutenFree,
            isSpicy = isSpicy,
            orderCount = orderCount ?: 0
        )
    }
}